### PR TITLE
Support instances destination type for transformation

### PR DESCRIPTION
**Pull request by vuhainguyen**
CDF-17675

## Description
Please describe the change you have made.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/transformations/common.py [hunk 1]
```diff
@@ -163,6 +188,37 @@ def __hash__(self) -> int:
...
+            stacklevel=2,
+        )
+        super().__init__(type="instances")
+        self.view_external_id = view_external_id
+        self.view_version = view_version
+        self.view_space_external_id = view_space_external_id
+        self.instance_space_external_id = instance_space_external_id
+
+    def __hash__(self) -> int:
+        return hash(
```
Comment (haakonvt) [inline cognite/client/data_classes/transformations/common.py:211]: Out of curiosity, why are you implementing dunder hash for all transformation data classes?

**Review by silvavelosa — APPROVED** [review]
LGTM ✔️

File: cognite/client/data_classes/transformations/common.py (hunk 1 above)
Comment (vuhainguyen) [reply cognite/client/data_classes/transformations/common.py:211]: Great question, I don't have much idea about it, they are added long time ago I think. Maybe it's used for encoding/decoding ? We can create a draft PR to remove all the hash methods to see if it breaks smthing? 😂

Comment (haakonvt) [reply cognite/client/data_classes/transformations/common.py:211]: Haha okey, I had a feeling 🤣

**Pull request merged**